### DISCUSSION TITLE
installation sous Windows pour les nuls

### DISCUSSION THREAD

**Discussion by Andarlo** [fra589/cn5X#39] [Q&A]
bonjour j'ai un petit problème quand je lance une invite de commande sous Windows 10 .
j'ai ce message l'ors de l'installation : 

C:\Users\a9kr99>pip install pyqt5
'pip' n’est pas reconnu en tant que commande interne
ou externe, un programme exécutable ou un fichier de commandes.

du coup j'ai télécharger sur le net le fichier de python et fait son installation et j'ai autoriser le path dans le system mais quand je fait un double clik sur cn5.py une fenêtre s'ouvre et disparait et rien ne ce passe ?

avez vous déjà eu ce problème ou je mis prend mal  ?

merci

**Comment by fra589** [comment]
Bonjour @Andarlo,  
  
Tu as bien téléchargé la dernière version à partir du site officiel https://www.python.org/ ?  

As tu relancé les 2 commandes :  
`pip install pyqt5`  
`pip install pyserial`
après l'installation de Python?
Ces 2 commandes se sont elles bien passées sans erreur ?

Si cn5X.py ouvre sa fenêtre, plante et se ferme, il à probablement émis un message d'erreur. Pour le voir, il faut passer par l'invite de commande.
Après avoir ouvert l'invite de commande, tapes :  
`cd <le répertoire ou tu as copié cn5X>`
`python cn5X.py`

Tu devrait voir apparaitre la sortie d'erreurs du programme qui permettra normalement de comprendre ce qui se passe.

@++;
Gauthier.

**Reply by Andarlo** [reply]
j'ai bien installer la version 3.10.2 et j'ai autorisé les variable d'environnement a l'installation.
étant au  taff je ne peut pas vous dire plus sur le rapport d'incident.

**Reply by Andarlo** [reply]
salut j'ai trouver problème de connexion internet .
par contre la carte brancher a nu sur le pc me donne ce rapport est ce que c'est normal ?

2022-02-23 14:59:37 : Info    : grblCom: Starting grblComSerial thread on COM4.
2022-02-23 14:59:37 : Info    : grblComSerial.run(): Running "grblComSerial" from thread #0xc10.
2022-02-23 14:59:37 : Info    : grblComSerial.__openComPort(): comPort COM4 open.
2022-02-23 14:59:38 : Info    : cn5X++ : Grbl initialisé.
2022-02-23 14:59:38 : Warning : grblComSerial.__mainLoop(): utf-8 decode error, buff=b'U\x94\xa5\x8a)b\xb920190605:Hh\x15a\xe94:XYZU]\r\n'
2022-02-23 14:59:38 : Warning : grblComSerial.__mainLoop(): utf-8 decode error, buff=b'T\xd3\x93\x82r\x82\x82b\x82r\x82\x82\x82b\x82r\x82\x82\x82b\x82r\x82\x82\x82\xea5\n'

**Reply by fra589** [reply]
Les "Warning" sont généralement du à des soucis de communication sur le port série.
Cela arrive généralement lors de la connexion avec des cartes non officielles (clones chinois à bas prix) lorsque la carte n'a pas fini d'initialiser la connexion et que le logiciel l’interroge.
Cela n'est généralement pas grave, cn5X traite ce cas de figure et prévient simplement qu'il à eu ce problème de communication.

**Reply by Andarlo** [reply]
ok merci

**Comment by Andarlo** [comment]
Je te partage des images de mon pupitre de cnc a mon travail si cela peut t'intéresser pour agencé les fenêtres ainsi que les sous menus.

**Comment by Andarlo** [comment]
![1645784984753](https://user-images.githubusercontent.com/99893235/155702093-24d88d63-90fd-426b-b3dc-0f1d41b1c85b.jpg)
![1645784984746](https://user-images.githubusercontent.com/99893235/155702102-f6f80485-bc5b-423b-b151-fb5ee7dbb65d.jpg)

**Comment by Andarlo** [comment]
![Sans titre](https://user-images.githubusercontent.com/99893235/155702520-1c65842f-e5ea-40e7-93da-0acae7670e23.png)
